src/Components/Body.tsx

```diff
@@ -1,558 +1,547 @@
+      }),
+    });
+  }
+
+  function renderItem(index: number, key: number): JSX.Element {
+    let dateKey = property.year;
+    let name = "";
+    let offset = 0;
+    if (props.bodyConfig.propertySelected == 0) {
+      dateKey = property.month;
+      name = new Date(0, index + 1, 0).toLocaleString(
+        globals.localeObj.localeString,
+        {
+          month: "long",
+        }
+      );
+    } else {
+      name = String(index + globals.minimumYearValue);
+      offset = globals.maximumYearValue;
+    }
+    return (
+      <Button
+        key={key}
+        onClick={() => handleClick(dateKey, index + offset)}
+        variant="text"
+        color="primary"
+        size="large"
+        className={globals.classes.bodyList}
+      >
+        {name}
+      </Button>
+    );
+  }
+
+  function renderScroll(): JSX.Element {
+    if (props.boxClass == "box-wide") {
+      if (props.bodyConfig.propertySelected == 1) {
+        return (
+          <Box mt={10}>
+            <div className={globals.classes.bodyListContainer}>
+              {" "}
+              <ReactList
+                itemRenderer={(index: any, key: any) => renderItem(index, key)}
+                length={globals.maximumYearValue - globals.minimumYearValue}
+                type="uniform"
+              />
+            </div>
+          </Box>
+        );
+      } else {
+        return (
+          <Box mt={10}>
+            <div className={globals.classes.bodyListContainer}>
+              {" "}
+              <ReactList
+                itemRenderer={(index: any, key: any) => renderItem(index, key)}
+                length={12}
+                type="uniform"
+              />
+            </div>
+          </Box>
+        );
+      }
+    } else {
+      return <div />;
+    }
+  }
+
+  function getTextTitle(): string {
+    if (props.index == 1) {
+      return "End Date";
+    } else {
+      return "Start Date";
+    }
+  }
+
+  function renderTabButton(title: string): JSX.Element {
+    return (
+      <Button
+        color="primary"
+        variant="text"
+        className={globals.classes.bodyTabButton}
+      >
+        {title}
+      </Button>
+    );
+  }
+
+  function renderTab(index: number): void {
+    if (index !== 0) {
+      props.setBoxClass("box-tiny");
+    } else {
+      props.setBoxClass("box");
+    }
+    props.setBodyConfig({ ...props.bodyConfig, bodySubTabIndex: index });
+  }
+
+  useEffect(() => {
+    props.setBodyConfig({ ...props.bodyConfig, bodySubTabIndex: 0 });
+    props.setBoxClass("box");
+  }, [props.index]);
+
+  useEffect(() => {
+    if (props.boxClass == "box-closed") {
+      props.setBodyConfig({ ...props.bodyConfig, bodySubTabIndex: 0 });
+    }
+  }, [props.boxClass]);
+
+  function getDateSelectObj() {
+    return {
+      setDropdownData: props.setDropdownData,
+      dropdownData: props.dropdownData,
+      applyFn,
+      dateRange: props.dateRange,
+      classes: globals.classes,
+    };
+  }
+
+  return (
```

Comment:
This desperately needs to be decomposed into a collection of smaller, tighter components, so that this becomes much simpler to read through.  That will also allow you to pull a bunch of the code and handlers above out into separate files.

Reply:
Done